core-services/src/main/java/org/zalando/nakadi/service/publishing/NakadiKpiPublisher.java

```diff
@@ -70,61 +74,51 @@ public void publish(final Supplier<KPIEvent> kpiEventSupplier) {
                 return;
             }
             final var kpiEvent = kpiEventSupplier.get();
-            final var eventType = kpiEvent.getName();
+            final var eventTypeName = kpiEvent.getName();
 
             if (featureToggleService.isFeatureEnabled(Feature.AVRO_FOR_KPI_EVENTS)) {
+                final var partition = metadataRandomPartitioner.calculatePartition(eventTypeName);
```

Comment:
I'd expect the partitioner for binary payloads to work on the same level as we already have for JSON.  That would allow us to reuse exiting code and keep the pieces of code related to partitioning closer together.

Reply:
I wanted to maintain a smaller scope of the PR and tackle this problem once the Publishing in Avro part is finished.

Reply:
Hm, maybe we should wait for that change to be merged first and base off it?  Otherwise I fear we'll have to throw away most of the changes here later.

Reply:
that code will be moved into publisher, once it is ready